File: app/src/main/java/com/j/jOS/egg/PlatLogoSnapshotProvider.java
package com.j.jOS.egg;

import static com.j.jOS.egg.PlatLogoActivity.BGCOLOR;

import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;


public class PlatLogoSnapshotProvider extends com.dede.basic.PlatLogoSnapshotProvider {

    @Override
    public boolean getIncludeBackground() {
        return true;
    }

    @Override
    public View create(Context context) {
        FrameLayout mContent = new FrameLayout(context);

        final FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.WRAP_CONTENT,
                FrameLayout.LayoutParams.WRAP_CONTENT);
        lp.gravity = Gravity.CENTER;

        final ImageView logo = new ImageView(context);
        logo.setImageResource(R.drawable.k_platlogo);
        logo.setScaleType(ImageView.ScaleType.CENTER_INSIDE);

        mContent.setBackgroundColor(BGCOLOR);
        mContent.addView(logo, lp);
        return mContent;
    }
}

File: commLib/java/com/dede/basic/PlatLogoSnapshotProvider.kt
package com.dede.basic

import android.content.Context
import android.view.View
import android.view.ViewGroup
import java.lang.ref.WeakReference

abstract class PlatLogoSnapshotProvider {

    @Deprecated
    open val includeBackground: Boolean = false

    @Deprecated
    private var cache: WeakReference<View>? = null

    @Deprecated
    fun get(context: Context): View {
        var view = cache?.get()
        if (view != null) {
            (view.parent as? ViewGroup)?.removeView(view)
            return view
        }
        view = create(context)
        cache = WeakReference(view)
        return view
    }

    @Deprecated
    abstract fun create(context: Context): View
}
